plugins/woocommerce/src/Blocks/BlockTypes/ProductQuery.php

```diff
@@ -180,9 +182,11 @@ public function update_query( $pre_render, $parsed_block ) {
 		if ( self::is_woocommerce_variation( $parsed_block ) ) {
 			// Indicate to interactivity powered components that this block is on the page
 			// and needs refresh to update data.
-			$this->asset_data_registry->add(
-				'needsRefreshForInteractivityAPI',
-				true
+			wp_interactivity_config(
+				'woocommerce',
+				[
+					'needsRefreshForInteractivityAPI' => true,
```

Comment:
ClassicTemplate block also registers this setting and need to be replaced.

Reply:
Thanks for spotting that 🙇

Reply:
Resolved in [a8504e9](https://github.com/woocommerce/woocommerce/pull/57469/commits/a8504e91c4561b75edf36ddf4fc9bf1ca93e93f6)